ClosedXML/Excel/Coordinates/XLAddress.cs

```diff
@@ -235,7 +241,7 @@ public string ToString(XLReferenceStyle referenceStyle, bool includeSheet)
 
             if (includeSheet)
                 return String.Concat(
-                    Worksheet.Name.EscapeSheetName(),
+                    WorksheetIsDeleted ? "#REF" : Worksheet.Name.EscapeSheetName(),
```

Comment:
When you fix the for when `includeSheet == true`, then if address is invalid and worksheet is deleted, the expected result should be only `#REF!`.